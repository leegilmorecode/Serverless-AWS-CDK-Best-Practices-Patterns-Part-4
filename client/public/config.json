{
  "stage": "featuredev",
  "domainName": "your-domain.co.uk",
  "subDomain": "featuredev.your-domain.co.uk",
  "api": "https://api-featuredev.your-domain.co.uk"
}
