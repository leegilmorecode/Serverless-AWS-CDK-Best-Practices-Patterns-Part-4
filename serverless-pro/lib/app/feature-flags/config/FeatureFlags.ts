export interface FeatureFlags {
  flags: {
    createOrderAllowList: {
      name: string;
      description: string;
      attributes: {
        allow: {
          constraints: {
            type: string;
            enum: Array<'beta-group' | 'qa'>;
            required: true;
          };
        };
      };
    };
    releaseCheckCreateOrderQuantity: {
      name: string;
      description: string;
      attributes: {
        limit: {
          constraints: {
            type: string;
            required: true;
          };
        };
      };
      _deprecation?: {
        status: string;
      };
    };
    opsPreventCreateOrders: {
      name: string;
      description: string;
    };
    opsLimitListOrdersResults: {
      name: string;
      description: string;
      attributes: {
        limit: {
          constraints: {
            type: string;
            required: true;
          };
        };
      };
    };
  };
  values: {
    createOrderAllowList: {
      enabled: boolean;
      allow: 'beta-group' | 'qa';
    };
    releaseCheckCreateOrderQuantity: {
      enabled: boolean;
      limit: number;
    };
    opsPreventCreateOrders: {
      enabled: boolean;
    };
    opsLimitListOrdersResults: {
      enabled: boolean;
      limit: number;
    };
  };
  version: string;
}

export type Flags = keyof FeatureFlags['flags'];
