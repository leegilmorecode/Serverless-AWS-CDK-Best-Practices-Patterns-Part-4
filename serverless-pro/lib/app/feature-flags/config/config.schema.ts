export const schema = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  definitions: {
    flagSetDefinition: {
      type: 'object',
      properties: {
        version: {
          $ref: '#/definitions/flagSchemaVersions',
        },
        flags: {
          $ref: '#/definitions/flagDefinitions',
        },
        values: {
          $ref: '#/definitions/flagValues',
        },
      },
      required: ['version', 'flags'],
      additionalProperties: false,
    },
    flagDefinitions: {
      type: 'object',
      patternProperties: {
        '^[a-z][a-zA-Zd-]{0,63}$': {
          $ref: '#/definitions/flagDefinition',
        },
      },
      maxProperties: 100,
      additionalProperties: false,
    },
    flagDefinition: {
      type: 'object',
      properties: {
        name: {
          $ref: '#/definitions/customerDefinedName',
        },
        description: {
          $ref: '#/definitions/customerDefinedDescription',
        },
        _createdAt: {
          type: 'string',
        },
        _updatedAt: {
          type: 'string',
        },
        _deprecation: {
          type: 'object',
          properties: {
            status: {
              type: 'string',
              enum: ['planned'],
            },
          },
          additionalProperties: false,
        },
        attributes: {
          $ref: '#/definitions/attributeDefinitions',
        },
      },
      additionalProperties: false,
    },
    attributeDefinitions: {
      type: 'object',
      patternProperties: {
        '^[a-z][a-zA-Zd-]{0,63}$': {
          $ref: '#/definitions/attributeDefinition',
        },
      },
      maxProperties: 25,
      additionalProperties: false,
    },
    attributeDefinition: {
      type: 'object',
      properties: {
        description: {
          $ref: '#/definitions/customerDefinedDescription',
        },
        constraints: {
          oneOf: [
            { $ref: '#/definitions/numberConstraints' },
            { $ref: '#/definitions/stringConstraints' },
            { $ref: '#/definitions/arrayConstraints' },
            { $ref: '#/definitions/boolConstraints' },
          ],
        },
      },
      additionalProperties: false,
    },
    flagValues: {
      type: 'object',
      patternProperties: {
        '^[a-z][a-zA-Zd-]{0,63}$': {
          $ref: '#/definitions/flagValue',
        },
      },
      maxProperties: 100,
      additionalProperties: false,
    },
    flagValue: {
      type: 'object',
      properties: {
        enabled: {
          type: 'boolean',
        },
        _createdAt: {
          type: 'string',
        },
        _updatedAt: {
          type: 'string',
        },
      },
      patternProperties: {
        '^[a-z][a-zA-Zd-]{0,63}$': {
          $ref: '#/definitions/attributeValue',
          maxProperties: 25,
        },
      },
      required: ['enabled'],
      additionalProperties: false,
    },
    attributeValue: {
      oneOf: [
        { type: 'string', maxLength: 1024 },
        { type: 'number' },
        { type: 'boolean' },
        {
          type: 'array',
          oneOf: [
            {
              items: {
                type: 'string',
                maxLength: 1024,
              },
            },
            {
              items: {
                type: 'number',
              },
            },
          ],
        },
      ],
      additionalProperties: false,
    },
    stringConstraints: {
      type: 'object',
      properties: {
        type: {
          type: 'string',
          enum: ['string'],
        },
        required: {
          type: 'boolean',
        },
        pattern: {
          type: 'string',
          maxLength: 1024,
        },
        enum: {
          type: 'array',
          maxLength: 100,
          items: {
            oneOf: [
              {
                type: 'string',
                maxLength: 1024,
              },
              {
                type: 'integer',
              },
            ],
          },
        },
      },
      required: ['type'],
      not: {
        required: ['pattern', 'enum'],
      },
      additionalProperties: false,
    },
    numberConstraints: {
      type: 'object',
      properties: {
        type: {
          type: 'string',
          enum: ['number'],
        },
        required: {
          type: 'boolean',
        },
        minimum: {
          type: 'integer',
        },
        maximum: {
          type: 'integer',
        },
      },
      required: ['type'],
      additionalProperties: false,
    },
    arrayConstraints: {
      type: 'object',
      properties: {
        type: {
          type: 'string',
          enum: ['array'],
        },
        required: {
          type: 'boolean',
        },
        elements: {
          $ref: '#/definitions/elementConstraints',
        },
      },
      required: ['type'],
      additionalProperties: false,
    },
    boolConstraints: {
      type: 'object',
      properties: {
        type: {
          type: 'string',
          enum: ['boolean'],
        },
        required: {
          type: 'boolean',
        },
      },
      required: ['type'],
      additionalProperties: false,
    },
    elementConstraints: {
      oneOf: [
        { $ref: '#/definitions/numberConstraints' },
        { $ref: '#/definitions/stringConstraints' },
      ],
    },
    customerDefinedName: {
      type: 'string',
      pattern: '^[^\\n]{1,64}$',
    },
    customerDefinedDescription: {
      type: 'string',
      maxLength: 1024,
    },
    flagSchemaVersions: {
      type: 'string',
      enum: ['1'],
    },
  },
  type: 'object',
  $ref: '#/definitions/flagSetDefinition',
};
