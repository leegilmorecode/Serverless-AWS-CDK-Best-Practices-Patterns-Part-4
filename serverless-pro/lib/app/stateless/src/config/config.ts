import convict = require('convict');

export const config = convict({
  tableName: {
    doc: 'The orders table',
    default: '',
    env: 'TABLE_NAME',
    nullable: false,
  },
  bucketname: {
    doc: 'The s3 bucket which create orders uses to push new invoices to',
    default: '',
    env: 'BUCKET_NAME',
    nullable: false,
  },
  appConfig: {
    appConfigApplicationId: {
      doc: 'The appConfig application ID',
      default: '',
      env: 'APPCONFIG_APPLICATION_ID',
      nullable: false,
    },
    appConfigEnvironmentId: {
      doc: 'The appConfig environment ID',
      default: '',
      env: 'APPCONFIG_ENVIRONMENT_ID',
      nullable: false,
    },
    appConfigConfigurationId: {
      doc: 'The appConfig configuration ID',
      default: '',
      env: 'APPCONFIG_CONFIGURATION_ID',
      nullable: false,
    },
  },
  flags: {
    createOrderAllowList: {
      doc: 'The feature flag for create order allow list',
      default: '',
      env: 'FLAG_CREATE_ORDER_ALLOW_LIST',
      nullable: false,
    },
    preventCreateOrder: {
      doc: 'The feature flag for preventing the creation of orders',
      default: '',
      env: 'FLAG_PREVENT_CREATE_ORDERS',
    },
    checkCreateOrderQuantity: {
      doc: 'The feature flag release for checking the quantity limit on an order',
      default: '',
      env: 'FLAG_CHECK_CREATE_ORDER_QUANTITY',
    },
    limitOrdersList: {
      doc: 'The feature flag for operationally limiting the orders on a list',
      default: '',
      env: 'FLAG_LIMIT_LIST_ORDERS_RESULTS',
    },
  },
  shared: {
    ordersWebsiteUrl: {
      doc: 'The orders website URL',
      default: '',
      env: 'WEBSITE_URL',
      nullable: false,
    },
    ordersApiHost: {
      doc: 'The orders API host',
      default: '',
      env: 'APP_API_HOST',
      nullable: false,
    },
    stage: {
      doc: 'The current stage',
      default: 'develop',
      env: 'STAGE',
      nullable: false,
    },
    randomErrorsEnabled: {
      doc: 'A boolean to toggle the random throwing of errors',
      default: 'false',
      env: 'RANDOM_ERRORS_ENABLED',
    },
    functions: {
      logLevel: {
        doc: 'The log levels for the logger',
        default: 'INFO',
        env: 'LOG_LEVEL',
        nullable: false,
      },
      logEvent: {
        doc: 'Whether or not to log the incoming event',
        default: 'false',
        env: 'POWERTOOLS_LOGGER_LOG_EVENT',
        nullable: false,
      },
      logSampleRate: {
        doc: 'The sample rate of the logging between 0-1',
        default: 1,
        format: 'int',
        env: 'POWERTOOLS_LOGGER_SAMPLE_RATE',
        nullable: false,
      },
    },
  },
});
