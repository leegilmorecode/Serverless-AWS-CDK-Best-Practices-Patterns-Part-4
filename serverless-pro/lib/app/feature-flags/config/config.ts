// There are four typical main types of feature flags:
// release flags, operational flags, experimental flags, and customer/permission flags.
// https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html#appconfig-type-reference-feature-flags

import { FeatureFlags } from './FeatureFlags';
import { Stage } from '../../../types';

export type FeatureFlagConfig = Record<Stage, FeatureFlags>;

export const environments: FeatureFlagConfig = {
  // allow developers to spin up a quick branch for a given PR they are working on e.g. pr-124
  // this is done with a npm run develop, not through the pipeline, and uses the values in .env
  [Stage.develop]: {
    flags: {
      createOrderAllowList: {
        name: 'createOrderAllowList',
        description:
          'When enabled it limits the allow list to a select set of groups',
        attributes: {
          allow: {
            constraints: {
              type: 'string',
              enum: ['beta-group', 'qa'],
              required: true,
            },
          },
        },
      },
      releaseCheckCreateOrderQuantity: {
        name: 'releaseCheckCreateOrderQuantity',
        description:
          'A release flag for the create order check on max quantity',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
        _deprecation: {
          status: 'planned',
        },
      },
      opsPreventCreateOrders: {
        name: 'opsPreventCreateOrders',
        description: 'Operational toggle to prevent the creation of new orders',
      },
      opsLimitListOrdersResults: {
        name: 'opsLimitListOrdersResults',
        description: 'Operation toggle to limit the results on list orders',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
      },
    },
    values: {
      createOrderAllowList: {
        enabled: false,
        allow: 'qa',
      },
      releaseCheckCreateOrderQuantity: {
        enabled: true,
        limit: 10,
      },
      opsPreventCreateOrders: {
        enabled: false,
      },
      opsLimitListOrdersResults: {
        enabled: true,
        limit: 10,
      },
    },
    version: '1',
  },
  [Stage.featureDev]: {
    flags: {
      createOrderAllowList: {
        name: 'createOrderAllowList',
        description:
          'when enabled it limits the allow list to a select set of groups',
        attributes: {
          allow: {
            constraints: {
              type: 'string',
              enum: ['beta-group', 'qa'],
              required: true,
            },
          },
        },
      },
      releaseCheckCreateOrderQuantity: {
        name: 'releaseCheckCreateOrderQuantity',
        description:
          'A release flag for the create order check on max quantity',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
        _deprecation: {
          status: 'planned',
        },
      },
      opsPreventCreateOrders: {
        name: 'opsPreventCreateOrders',
        description: 'Operational toggle to prevent the creation of new orders',
      },
      opsLimitListOrdersResults: {
        name: 'opsLimitListOrdersResults',
        description: 'Operation toggle to limit the results on list orders',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
      },
    },
    values: {
      createOrderAllowList: {
        enabled: false,
        allow: 'qa',
      },
      releaseCheckCreateOrderQuantity: {
        enabled: true,
        limit: 1000,
      },
      opsPreventCreateOrders: {
        enabled: false,
      },
      opsLimitListOrdersResults: {
        enabled: true,
        limit: 1000,
      },
    },
    version: '1',
  },
  [Stage.staging]: {
    flags: {
      createOrderAllowList: {
        name: 'createOrderAllowList',
        description:
          'when enabled it limits the allow list to a select set of groups',
        attributes: {
          allow: {
            constraints: {
              type: 'string',
              enum: ['beta-group', 'qa'],
              required: true,
            },
          },
        },
      },
      releaseCheckCreateOrderQuantity: {
        name: 'releaseCheckCreateOrderQuantity',
        description:
          'A release flag for the create order check on max quantity',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
        _deprecation: {
          status: 'planned',
        },
      },
      opsPreventCreateOrders: {
        name: 'opsPreventCreateOrders',
        description: 'Operational toggle to prevent the creation of new orders',
      },
      opsLimitListOrdersResults: {
        name: 'opsLimitListOrdersResults',
        description: 'Operation toggle to limit the results on list orders',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
      },
    },
    values: {
      createOrderAllowList: {
        enabled: true,
        allow: 'qa',
      },
      releaseCheckCreateOrderQuantity: {
        enabled: true,
        limit: 1000,
      },
      opsPreventCreateOrders: {
        enabled: false,
      },
      opsLimitListOrdersResults: {
        enabled: true,
        limit: 1000,
      },
    },
    version: '1',
  },
  [Stage.prod]: {
    flags: {
      createOrderAllowList: {
        name: 'createOrderAllowList',
        description:
          'when enabled it limits the allow list to a select set of groups',
        attributes: {
          allow: {
            constraints: {
              type: 'string',
              enum: ['beta-group', 'qa'],
              required: true,
            },
          },
        },
      },
      releaseCheckCreateOrderQuantity: {
        name: 'releaseCheckCreateOrderQuantity',
        description:
          'A release flag for the create order check on max quantity',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
        _deprecation: {
          status: 'planned',
        },
      },
      opsPreventCreateOrders: {
        name: 'opsPreventCreateOrders',
        description: 'Operational toggle to prevent the creation of new orders',
      },
      opsLimitListOrdersResults: {
        name: 'opsLimitListOrdersResults',
        description: 'Operation toggle to limit the results on list orders',
        attributes: {
          limit: {
            constraints: {
              type: 'number',
              required: true,
            },
          },
        },
      },
    },
    values: {
      createOrderAllowList: {
        enabled: false,
        allow: 'qa',
      },
      releaseCheckCreateOrderQuantity: {
        enabled: false,
        limit: 1000,
      },
      opsPreventCreateOrders: {
        enabled: false,
      },
      opsLimitListOrdersResults: {
        enabled: false,
        limit: 1000,
      },
    },
    version: '1',
  },
};
