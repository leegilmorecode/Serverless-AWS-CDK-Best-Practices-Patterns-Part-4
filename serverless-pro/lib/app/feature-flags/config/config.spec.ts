/* eslint-disable @typescript-eslint/no-var-requires */
// https://github.com/cypress-io/cypress/issues/22059
import { describe, expect, it } from '@jest/globals';

import Ajv from 'ajv';
import { environments } from './config';

const ajv = new Ajv({ strict: false });

describe('feature flag configuration', () => {
  describe('develop stage i.e. ephemeral', () => {
    it('should match the snapshot', () => {
      expect(environments.develop).toMatchSnapshot();
    });

    it('should validate with the schema', () => {
      const { schema } = require('./config.schema');
      expect(ajv.validate(schema, environments.develop)).toEqual(true);
    });
  });

  describe('featureDev stage', () => {
    it('should match the snapshot', () => {
      expect(environments.featureDev).toMatchSnapshot();
    });

    it('should validate with the schema', () => {
      const { schema } = require('./config.schema');
      expect(ajv.validate(schema, environments.featureDev)).toEqual(true);
    });
  });

  describe('staging stage', () => {
    it('should match the snapshot', () => {
      expect(environments.staging).toMatchSnapshot();
    });

    it('should validate with the schema', () => {
      const { schema } = require('./config.schema');
      expect(ajv.validate(schema, environments.staging)).toEqual(true);
    });
  });

  describe('prod stage', () => {
    it('should match the snapshot', () => {
      expect(environments.prod).toMatchSnapshot();
    });

    it('should validate with the schema', () => {
      const { schema } = require('./config.schema');
      expect(ajv.validate(schema, environments.prod)).toEqual(true);
    });
  });
});
