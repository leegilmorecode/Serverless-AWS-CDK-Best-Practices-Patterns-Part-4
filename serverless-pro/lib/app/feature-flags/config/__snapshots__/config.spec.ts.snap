// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature flag configuration develop stage i.e. ephemeral should match the snapshot 1`] = `
{
  "flags": {
    "createOrderAllowList": {
      "attributes": {
        "allow": {
          "constraints": {
            "enum": [
              "beta-group",
              "qa",
            ],
            "required": true,
            "type": "string",
          },
        },
      },
      "description": "When enabled it limits the allow list to a select set of groups",
      "name": "createOrderAllowList",
    },
    "opsLimitListOrdersResults": {
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "Operation toggle to limit the results on list orders",
      "name": "opsLimitListOrdersResults",
    },
    "opsPreventCreateOrders": {
      "description": "Operational toggle to prevent the creation of new orders",
      "name": "opsPreventCreateOrders",
    },
    "releaseCheckCreateOrderQuantity": {
      "_deprecation": {
        "status": "planned",
      },
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "A release flag for the create order check on max quantity",
      "name": "releaseCheckCreateOrderQuantity",
    },
  },
  "values": {
    "createOrderAllowList": {
      "allow": "qa",
      "enabled": false,
    },
    "opsLimitListOrdersResults": {
      "enabled": true,
      "limit": 10,
    },
    "opsPreventCreateOrders": {
      "enabled": false,
    },
    "releaseCheckCreateOrderQuantity": {
      "enabled": true,
      "limit": 10,
    },
  },
  "version": "1",
}
`;

exports[`feature flag configuration featureDev stage should match the snapshot 1`] = `
{
  "flags": {
    "createOrderAllowList": {
      "attributes": {
        "allow": {
          "constraints": {
            "enum": [
              "beta-group",
              "qa",
            ],
            "required": true,
            "type": "string",
          },
        },
      },
      "description": "when enabled it limits the allow list to a select set of groups",
      "name": "createOrderAllowList",
    },
    "opsLimitListOrdersResults": {
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "Operation toggle to limit the results on list orders",
      "name": "opsLimitListOrdersResults",
    },
    "opsPreventCreateOrders": {
      "description": "Operational toggle to prevent the creation of new orders",
      "name": "opsPreventCreateOrders",
    },
    "releaseCheckCreateOrderQuantity": {
      "_deprecation": {
        "status": "planned",
      },
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "A release flag for the create order check on max quantity",
      "name": "releaseCheckCreateOrderQuantity",
    },
  },
  "values": {
    "createOrderAllowList": {
      "allow": "qa",
      "enabled": false,
    },
    "opsLimitListOrdersResults": {
      "enabled": true,
      "limit": 1000,
    },
    "opsPreventCreateOrders": {
      "enabled": false,
    },
    "releaseCheckCreateOrderQuantity": {
      "enabled": true,
      "limit": 1000,
    },
  },
  "version": "1",
}
`;

exports[`feature flag configuration prod stage should match the snapshot 1`] = `
{
  "flags": {
    "createOrderAllowList": {
      "attributes": {
        "allow": {
          "constraints": {
            "enum": [
              "beta-group",
              "qa",
            ],
            "required": true,
            "type": "string",
          },
        },
      },
      "description": "when enabled it limits the allow list to a select set of groups",
      "name": "createOrderAllowList",
    },
    "opsLimitListOrdersResults": {
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "Operation toggle to limit the results on list orders",
      "name": "opsLimitListOrdersResults",
    },
    "opsPreventCreateOrders": {
      "description": "Operational toggle to prevent the creation of new orders",
      "name": "opsPreventCreateOrders",
    },
    "releaseCheckCreateOrderQuantity": {
      "_deprecation": {
        "status": "planned",
      },
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "A release flag for the create order check on max quantity",
      "name": "releaseCheckCreateOrderQuantity",
    },
  },
  "values": {
    "createOrderAllowList": {
      "allow": "qa",
      "enabled": false,
    },
    "opsLimitListOrdersResults": {
      "enabled": false,
      "limit": 1000,
    },
    "opsPreventCreateOrders": {
      "enabled": false,
    },
    "releaseCheckCreateOrderQuantity": {
      "enabled": false,
      "limit": 1000,
    },
  },
  "version": "1",
}
`;

exports[`feature flag configuration staging stage should match the snapshot 1`] = `
{
  "flags": {
    "createOrderAllowList": {
      "attributes": {
        "allow": {
          "constraints": {
            "enum": [
              "beta-group",
              "qa",
            ],
            "required": true,
            "type": "string",
          },
        },
      },
      "description": "when enabled it limits the allow list to a select set of groups",
      "name": "createOrderAllowList",
    },
    "opsLimitListOrdersResults": {
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "Operation toggle to limit the results on list orders",
      "name": "opsLimitListOrdersResults",
    },
    "opsPreventCreateOrders": {
      "description": "Operational toggle to prevent the creation of new orders",
      "name": "opsPreventCreateOrders",
    },
    "releaseCheckCreateOrderQuantity": {
      "_deprecation": {
        "status": "planned",
      },
      "attributes": {
        "limit": {
          "constraints": {
            "required": true,
            "type": "number",
          },
        },
      },
      "description": "A release flag for the create order check on max quantity",
      "name": "releaseCheckCreateOrderQuantity",
    },
  },
  "values": {
    "createOrderAllowList": {
      "allow": "qa",
      "enabled": true,
    },
    "opsLimitListOrdersResults": {
      "enabled": true,
      "limit": 1000,
    },
    "opsPreventCreateOrders": {
      "enabled": false,
    },
    "releaseCheckCreateOrderQuantity": {
      "enabled": true,
      "limit": 1000,
    },
  },
  "version": "1",
}
`;
